src/main/java/nl/smallproject/www/techiteasy/services/TelevisionService.java

```diff
@@ -58,14 +87,79 @@ public void deleteTelevision(Long id) {
         televisionRepository.deleteById(id);
     }
 
-    public List<TelevisionOutputDto> getAllTelevision() {
-        List<Television> televisions = televisionRepository.findAll();
-        List<TelevisionOutputDto> televisionOutputDtos = new ArrayList<>();
+    public void assignRemoteControllerToTelevision(Long televisionId, Long remoteControllerId) {
+        Optional<Television> televisionOptional = Optional.ofNullable(televisionRepository.findById(televisionId)
+                .orElseThrow(() -> new RecordNotFoundException("Television not found with this id: " + televisionId)));
 
-        for (var television: televisions) {
-            televisionOutputDtos.add(televisionMapper.televisionEntityToOutputDto(television));
+        Optional<RemoteController> remoteControllerOptional = Optional.ofNullable(remoteControllerRepository.findById(remoteControllerId)
```

Comment:
Je hoeft niet een optional te maken door "Optional.ofNullable"  te gebruiken. De findbyid methode geeft van zichzelf al een Optional terug. Dus wat je nu doet is dat je een optional ophaalf uit de database, met "orelsethrow"  checkt of er wat in zit (zo niet gooi je een exception op) en daarna zet je die uitkomst in een Optional zodat je hieronder nog een keer met isPresent de optional moet checken.